packages/vite/src/node/plugins/html.ts

```diff
@@ -588,7 +588,7 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
           }
         }
 
-        const shortEmitName = path.posix.relative(config.root, id)
+        const shortEmitName = path.relative(config.root, id)
```

Comment:
We also need to re-normalize the url (or posix-ify) it on this line

https://github.com/vitejs/vite/blob/a50601d98bce895d713c55911471db69f0cf0c44/packages/vite/src/node/plugins/html.ts#L609

Reply:
I'm not sure I understand. Isn't `shortEmitName` already in the desired format for this, as POSIX-style paths match URL-style?

Reply:
Yeah we need URL-style here because the `path` property points to the public path URL:

https://github.com/vitejs/vite/blob/7f8dc58daa4a9f07c71d069ecf1f2b055521f23f/packages/vite/src/node/plugins/html.ts#L660-L663

Otherwise with this PR, we get `path: "/foo\\bar.html"` on Windows, which wouldn't work if used in URLs.

Reply:
Oh, yes, sorry! I misread my own PR and thought the .posix was still there.

I'll add it back in for the URL, when I have some time. 👍